### PR TITLE
[HDRP]Framework to point to documentation for sections

### PR DESCRIPTION
**Pull request by alex-vazquez**
---
### Purpose of this PR

Adding documentation links to Light Sections

image

---
### Testing status
Check the links opens the browser on the given sections

---
### Comments to reviewers
Adding a way to declare enums with `HelpURLAttribute` this allow us to get it on the FoldoutGroup and being able to add the documentation icon generically.

### CODE REVIEW

File: com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs [hunk 1]
```diff
@@ -534,6 +537,20 @@ void IDrawer.Draw(TData data, Editor o)
...
+        {
+            var helpURLAttribute = (HelpURLAttribute)mask
+                .GetType()
+                .GetCustomAttributes(typeof(HelpURLAttribute), false)
+                .FirstOrDefault();
+
+            if (helpURLAttribute == null)
+                return string.Empty;
+
+            return $"{helpURLAttribute.URL}#{mask}";
```
Comment (RSlysz) [inline com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs:551]: ~~Is doc is ok to have this anchor in doc pages?~~
Well seams it was already the case

**Review by RSlysz — APPROVED** [review]

**Comment by alex-vazquez** [comment]
@Unity-Technologies/gfx-qa-hdrp , could you please test the links are opening the right sections?

Can you go to camera and see that there are no links( for now it is not implemented on the camera inspector, but I would like to make sure that if no documentation is defined this is not crashing )

**Comment by JMargevics** [comment]
Glad to finally see doc links! 
Will list here issues and missing links that I've found. 

1) Looks like Light sub category links aren't generated correctly. Example: **HD Additional Light Data** leads to https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@12.0/manual/Light-Component.html which is correct but if you try shape link it leads to https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@11.0/manual/Light-Component.html.html#Shape where we have double .html

image
2) For some reason [Raytrace SSS](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@11.0/manual/Ray-Traced-Subsurface-Scattering.html) and settings component doc buttons are missing. However I can't find any doc page for Ray Tracing Settings component, maybe @JordanL8 could comment? 

3) Local Volumetric Fog component leads to https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@11.0/manual/Local-Volumetric-Fog.html but in actuality doc page is still called [Density Volume](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@11.0/manual/Density-Volume.html). Maybe again more related to @JordanL8 :) 

4) A suggestion: It would super cool to add similar buttons to our tools with separate windows: LookDev, HDRP Wizard, Graph viewer, Debugger. 

For example Compositor already got one: 
image

5) Also wondering if it's possible to add a link to [Frame Settings
](https://docs.unity3d.com/Packages/com.unity.render-pipelines.high-definition@11.0/manual/HDRP-Camera.html)
Somewhere in camera and default settings? 
image

**Comment by alex-vazquez** [comment]
@JMargevics thanks for your deep QA pass. 

As this PR only touches lights, could you create a fogbugz and assign it to me with the other errors that you found?

The double .html is solved.

**Review by JMargevics — APPROVED** [review]
All light subcategory links are being generated correctly now. Approving the PR ✅ 

Will report missing raytracing buttons to Fogbugz 👍

File: com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs [hunk 2]
```diff
@@ -554,9 +571,12 @@ void IDrawer.Draw(TData data, Editor o)
                 else
                 {
                     CoreEditorUtils.DrawSplitter(isBoxed);
-                    newExpended = CoreEditorUtils.DrawHeaderFoldout(title, expended, isBoxed,
+                    newExpended = CoreEditorUtils.DrawHeaderFoldout(title,
+                        expended,
```
Comment (jenniferd-unity) [inline com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs:575]: could we rename with an 'a' ? should probably be expanded and not expended :)

Comment (jenniferd-unity) [reply com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs:575]: doesnt have to be this PR :)

**Review by jenniferd-unity — APPROVED** [review]

File: com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs (hunk 2 above)
Comment (alex-vazquez) [reply com.unity.render-pipelines.core/Editor/CoreEditorDrawers.cs:575]: I will do :)

**Pull request merged**